scripts/viz/param_sweep_viz.R

```diff
@@ -1,101 +1,209 @@
+      facet_grid(as.formula(paste("batch_size ~", facet_var)),
+                 scales = "free") +
+      theme_bw() + 
+      ggtitle(paste(param, "=", v))
+    
+    output_fig <- paste0(base_file, v, "_", param, ".png")
+    ggsave(output_fig, plot = p, height = 5, width = 6)
+  }
+}
+
+plotFinalParams <- function(df, base_file, twohidden = FALSE) {
+  # Plot the final validation loss at the end of training
+  #
+  # Arguments:
+  # df - dataframe that stores validation loss at training end across params
+  # base_file - directory and basename of where file will be saved
+  # twohidden - boolean to determine how variables are displayed
+  #
+  # Output:
+  # Saves the resulting plot to file
+
+  if (twohidden) {
+    # Recode epochs variable for plotting
+    df$epochs <-  dplyr::recode_factor(df$epochs, 
+                           "epochs: 10" = 10, 
+                           "epochs: 25" = 25, 
+                           "epochs: 50" = 50, 
+                           "epochs: 100" = 100) 
+    p <- ggplot(df, aes(x = epochs, y = end_loss)) +
+      geom_point(aes(shape = batch_size, color = learning_rate),
+                 size = 2) + facet_grid(~ kappa) +
+      xlab("Epochs")
+    height <- 5
+    width <- 10
+  } else {
+    df <- df %>% dplyr::filter(kappa == "kappa: 1.0")
+    p <- ggplot(df, aes(x = learning_rate, y = end_loss)) +
+      geom_point(aes(color = batch_size, shape = epochs), size = 2) +
+      xlab("Learning Rate")
+    height <- 5
+    width <- 6
+  }
+
+  p <- p + theme_bw() +
+    ylab("Validation Loss at Training End") +
+    theme(axis.text = element_text(size = rel(1.2)),
+          legend.text = element_text(size = rel(1.1)))
+
+  output_fig <- paste0(base_file, "final_param_val_loss.png")
+  ggsave(output_fig, plot = p, height = height, width = width)
 }
 
-final_select_df <- param_melt_df %>% dplyr::filter(loss_type == "val_loss")
-final_select_df <- final_select_df %>%
-  dplyr::group_by(learning_rate, batch_size, epochs) %>%
-  dplyr::summarize(end_loss = tail(loss, 1))
+# Set File names
+param_sweep_file <- file.path("results", "parameter_sweep_full_results.tsv")
+param_sweep_two_file <- file.path("results",
+                                  "parameter_sweep_twohidden_full_results.tsv")
+out_fig <- file.path("figures", "param_sweep", "full_param_")
+out_two_fig <- file.path("figures", "param_sweep", "twohidden", "full_param_")
+
+# Process files and plot data
+tybalt_df <- recodeParameters(param_sweep_file, depth = 1)
+tybalt_twohidden_df <- recodeParameters(param_sweep_two_file, depth = 2)
+
+plotSweep(tybalt_df[[1]], "kappa", out_fig)
+plotSweep(tybalt_twohidden_df[[1]], "kappa", out_two_fig)
+
+plotSweep(tybalt_df[[1]], "epochs", out_fig)
+plotSweep(tybalt_twohidden_df[[1]], "epochs", out_two_fig)
+
+plotFinalParams(tybalt_df[[2]], out_fig)
+plotFinalParams(tybalt_twohidden_df[[2]], twohidden = TRUE, out_two_fig)
+
+# Compare two hidden with one hidden layer
+best_one_layer <- tybalt_df[[1]] %>%
```

Comment:
You might consider not hard-coding the filters -- can you identify the "best" models in this script somewhere and use that to filter?